<launch>
  <!-- Arguments -->
    <arg name="node_id"/>
    <arg name="node_type"/>
    <arg name="rabbitmq_endpoint"/>
    <arg name="rabbitmq_username"/>
    <arg name="rabbitmq_password"/>
    <arg name="rabbitmq_port"/>
    <arg name="secret"/>
     <arg name="base_directory"/>

  <!--Robot namespace-->
    <group ns="$(arg node_id)">
        <!-- Load the controller -->
        <node pkg="multirobot_sim" type="roschain.py" name="roschain" output="screen" required="true">
        <param name="node_id" value="$(arg node_id)"/>
        <param name="node_type" value="$(arg node_type)"/>
        <param name="rabbitmq_endpoint" value="$(arg rabbitmq_endpoint)"/>
        <param name="rabbitmq_username" value="$(arg rabbitmq_username)"/>
        <param name="rabbitmq_password" value="$(arg rabbitmq_password)"/>
        <param name="rabbitmq_port" value="$(arg rabbitmq_port)"/>
        <param name="secret" value="$(arg secret)"/>
        <param name="base_directory" value="$(arg base_directory)"/>
        </node>  
    </group>
</launch>
